secrets/testing_test.go

```diff
@@ -205,6 +206,39 @@ func TestNewTestSecrets(t *testing.T) {
 	}
 }
 
+func TestRaceConditionFlag(t *testing.T) {
```

Comment:
This test fails when running `make test` without the accompanying changes because it triggers the race condition check.